src/components/timeline/form/TimelineDetailsPanel.svelte

```diff
@@ -0,0 +1,337 @@
+  function transformDraggedElement(draggedEl: Element) {
+    const el = draggedEl.querySelector('.row') as HTMLElement;
+    el.style.background = 'var(--st-gray-10)';
+    el.classList.add('test');
+  }
+
+  onMount(() => {
+    if ($selectedTimelineId === null) {
+      const firstTimeline = $view.definition.plan.timelines[0];
+      if (firstTimeline) {
+        viewSetSelectedTimeline(firstTimeline.id);
+      }
+    }
+  });
+</script>
+
+<Panel borderLeft padBody={false}>
+  <svelte:fragment slot="header">
+    <GridMenu {gridId} title="Timeline Details" />
+  </svelte:fragment>
+
+  <svelte:fragment slot="body">
+    <!-- Select Timeline. -->
+    <div class="timeline-select-container">
+      <select
+        class="st-select w-100"
+        data-type="number"
+        name="timelines"
+        value={$selectedTimelineId}
+        on:change={e => {
+          const { valueAsNumber: id } = getTarget(e);
+          viewSetSelectedTimeline(id);
+        }}
+      >
+        {#each $view.definition.plan.timelines as timeline}
+          <option value={timeline.id}>
+            Timeline {timeline.id}
+          </option>
+        {/each}
+      </select>
+    </div>
+
+    <!-- Timeline -->
+    {#if !$selectedTimeline}
+      <fieldset class="editor-section">No timeline selected</fieldset>
+    {:else}
+      <fieldset class="editor-section">
+        <div class="st-typography-medium editor-section-header">Margins</div>
+        <CssGrid columns="1fr 1fr" gap="8px">
+          <Input>
+            <label for="marginLeft">Margin Left</label>
+            <input
+              class="st-input w-100"
+              name="marginLeft"
+              type="number"
+              value={$selectedTimeline.marginLeft}
+              on:input={updateTimelineEvent}
+            />
+          </Input>
+
+          <Input>
+            <label for="marginRight">Margin Right</label>
+            <input
+              class="st-input w-100"
+              name="marginRight"
+              type="number"
+              value={$selectedTimeline.marginRight}
+              on:input={updateTimelineEvent}
+            />
+          </Input>
+        </CssGrid>
+      </fieldset>
+
+      <fieldset class="editor-section">
+        <div class="editor-section-header editor-section-header-with-button">
+          <div class="st-typography-medium">Vertical Guides</div>
+          <button
+            on:click={handleNewGuideClick}
+            use:tooltip={{ content: 'New Vertical Guide', placement: 'top' }}
+            class="st-button icon"
+          >
+            <PlusIcon />
+          </button>
+        </div>
+        {#if verticalGuides.length}
+          <div>
+            {#each verticalGuides as verticalGuide (verticalGuide.id)}
+              <div class="vertical-guide">
+                <!-- TODO how do we prevent users from typing in out of bounds date ranges? -->
+                <DatePickerField
```

Comment:
@duranb Does the DatePicker currently have a way to resolve this TODO?

Reply:
Currently there's no logic to prevent them from typing it in, just logic to prevent them from clicking on dates out of the range I think. Generally, I've found it difficult to perfectly prevent users from typing incorrect things without running into weird edge cases. I've also found it to be a little confusing/misleading if we allow them to type something and then changing the value to something different and valid after they've entered it. I usually show just show validation errors and rely on the user to correct their mistake. That way, they're aware of it.

Reply:
In that case @AaronPlave can we remove this TODO?

Reply:
Can do